stumpy/core.py

```diff
@@ -2254,6 +2254,7 @@ def preprocess_diagonal(
 
     M_T, Σ_T = compute_mean_std(T, m)
     Σ_T[T_subseq_isconstant] = 1.0  # Avoid divide by zero in next inversion step
+    Σ_T[~T_subseq_isfinite] = 1.0
```

Comment:
@seanlaw 

I was thinking more about this and noticed that `Σ_T[~T_subseq_isfinite] = 1.0` sets the std of ANY non-finite subsequence to `1.0`. For example, the std of subsequence `[np.nan, 1, 2]` will be set to `1.0`. This however does not affect the computation of matrix profile.

Reply:
@NimaSarajpoor I don't think I am getting your point. Is doing `Σ_T[~T_subseq_isfinite] = 1.0` a problem? To some extent, I kind of view ANY subsequence with a single non-finite value to be interpreted as a subsequence with ALL `np.nan`. Then, by that logic (maybe you can disagree?), that entire subsequence is "constant". Of course, it's not really constant! In other words, a single `np.inf`/`np.nan` within the subsequence renders the entire subsequence useless for computing distances and a "bad" subsequence cannot actually have a `Σ_T` value. What do you think?

Hmmm, now that I think about it, what happens if we set `Σ_T[~T_subseq_isfinite] = np.nan` instead?? This would avoid any divide-by-zero since it would be divide-by-nan, which is fine, and it would be separate/independent from `T_subseq_isconstant` and we keep things explicit?

Reply:
> Is doing `Σ_T[~T_subseq_isfinite] = 1.0` a problem?

Not really. I think we can do `Σ_T[~T_subseq_isfinite] = v`, where `v` is any non-zero value and we are still fine. In fact, we can do  `Σ_T[T_subseq_isconstant] = v` as well where `v` is just any non-zero value.

> what happens if we set Σ_T[~T_subseq_isfinite] = np.nan instead?? This would avoid any divide-by-zero since it would be divide-by-nan, which is fine, and it would be separate/independent from T_subseq_isconstant and we keep things explicit?

Right!! I think 

```
Σ_T[T_subseq_isconstant] = 1.0
Σ_T[~T_subseq_isfinite] = np.nan
```
 is better as it separates the "subsequence with at least one `np.nan`" case from the "constant, finite subsequence" case.

Reply:
Cool! Please let me know if this breaks any tests